build.sbt

```diff
@@ -101,5 +101,6 @@ dependencyOverrides ++= Seq(
   "com.fasterxml.jackson.core" % "jackson-core" % jacksonVersion,
   "software.amazon.awssdk" % "netty-nio-client" % "2.20.26",
   "org.json" % "json" % "20230227",
-  "io.netty" % "netty-handler" % "4.1.93.Final"
+  "io.netty" % "netty-handler" % "4.1.93.Final",
+  "org.xerial.snappy" % "snappy-java" % "1.1.10.1"
```

Comment:
Do you know where this was being introduced from? Is there an upgrade path for the source of that or is the override the only way out?

Reply:
Thanks Justin for asking, this is originated in sub-branch of the dependency - ` "software.amazon.kinesis" % "amazon-kinesis-client" % "2.5.0"`. The recommended solution by snyk is via `snappy-java` library which I did but this only when I already tried for upgrading `kinesis` client directly which is not able to pick by IDE so I kept kinesis version to 2.5.0 only and added override for snappy-java library.

Reply:
I searched for it now, `amazon-kinesis-client` has newest version releases just a week before that is on 27th June 23, the version is 2.5.1. Not sure if I can try this upgrade? need to confirm the testing as it is direct kinesis dependency upgrade.

Reply:
Update is, We have to move back to same change that is already here because

a) Tested **without** override `snappy-java` and doing `kinesis` version upgrade to latest `2.5.1` : 
-> so, this does not solve High snyk of "Denial of Service in snappy-java" library in firehose itself.
Hence, we have to do override snappy-java lib here. 
Changes will remain same in this firehose PR. 

b) Tested **with** override `snappy-java` and tested on `pubflow` with snapshot release of firehose that has got override `snappy-java`:
-> so, this does not resolve `pubflow`'s High snyk vuln for snappy-java as well. 
Hence, we have to override snappy-java lib in pubflow too.
Changes will remain same in Pubflow PR too.

Reply:
Also, I have checked that dependency overrides are if present in firehose then they all are same way present in pubflow too which contains firehose as direct dependency. so we have to have snappy-java as override in pubflow as well.

Reply:
Thanks Divya, it sounds like the specific override is the only viable fix for the time being. Thanks for checking, will approve.